### PR TITLE
Added images to brush pencils and changed structure

### PR DESCRIPTION
**Pull request by alexandrehcdc**

### CODE REVIEW

File: StantDrawing/BrushTool/BrushToolView.swift [hunk 1]
```diff
@@ -247,14 +248,24 @@ public class BrushToolView: UIView, GroupSizeContract {
         button.imageEdgeInsets = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
     }
     
-    private func setButtonImage(button: UIButton, imageName: String, imageColor: String) {
+    private func setButtonImage(button: Button, imageName: String, imageColor: String) {
+        let uiButton = button.uiButton
+
         let selectColorIcon      = UIImage(named: imageName, in: bundle, compatibleWith: nil)
         let selectColorImageView = UIImageView()
+        if button.imageSize != nil {
```
Comment (liuvasconcelos) [inline StantDrawing/BrushTool/BrushToolView.swift]: Why did you not use if let in this case? Just to keep the pattern.

File: StantDrawing/BrushTool/BrushToolView.swift [hunk 2]
```diff
@@ -247,14 +248,24 @@ public class BrushToolView: UIView, GroupSizeContract {
...
-        
-        button.tintColor = UIColor(hex: imageColor)
+
+        uiButton.setImage(selectColorImageView.image?.withRenderingMode(.alwaysTemplate), for: .normal)
+
+        uiButton.tintColor = UIColor(hex: imageColor)
+
+        guard let imageSize = button.imageSize else { return }
+
+        uiButton.imageView?.frame.size = imageSize
```
Comment (liuvasconcelos) [inline StantDrawing/BrushTool/BrushToolView.swift]: Actually, you can bring that setting of the other comment to here :)

File: StantDrawing/BrushTool/GroupSizeView.swift [hunk 3]
```diff
@@ -55,19 +55,51 @@ public class GroupSizeView: UIView {
         loadGroupPencilSizeButtons()
         
         configureGroupSizeLayout(groupButtons: groupSizeButtons, groupSizeView: self, iconReferenceName: iconReferenceName, heightBaseToCenter: heightBaseToCenter, xBaseToCenter: xBaseToCenter)
+
```
Comment (liuvasconcelos) [inline StantDrawing/BrushTool/GroupSizeView.swift:58]: Remove this extra lines.

File: StantDrawing/BrushTool/GroupSizeView.swift [hunk 4]
```diff
@@ -55,19 +55,51 @@ public class GroupSizeView: UIView {
...
-        let buttonSize2   = buildButton(buttonName: "2", sizeTool: 5, buttonWidth: 18, buttonHeight: 18)
-        let buttonSize3   = buildButton(buttonName: "3", sizeTool: 10, buttonWidth: 23, buttonHeight: 23)
-        let buttonSize4   = buildButton(buttonName: "4", sizeTool: 30, buttonWidth: 28, buttonHeight: 28)
-        let buttonSize5   = buildButton(buttonName: "5", sizeTool: 55, buttonWidth: 33, buttonHeight: 33)
-        let buttonSize6   = buildButton(buttonName: "6", sizeTool: 80, buttonWidth: 37, buttonHeight: 37)
+        let buttonSize1 = buildButton(buttonName: "1",
+                                      sizeTool: 2,
+                                      buttonWidth: 37,
+                                      buttonHeight: 37,
+                                      imageWidth: 13,
```
Comment (liuvasconcelos) [inline StantDrawing/BrushTool/GroupSizeView.swift:68]: I don't know if this function "buildButton" is used in another place with other usage. But, if it is used just here, try to use this buttonWidth and ButtonHeight with a default value in fuction (37, in this case) and pass the size of image in parameter. This way you will remove at least 3 parameters on function.

File: StantDrawing/BrushTool/GroupSizeView.swift [hunk 5]
```diff
@@ -133,25 +169,36 @@ public class GroupSizeView: UIView {
     private func setButtonsAsNotClicked(buttons: [Button]) {
         buttons.forEach { button in
             setButtonSizeAsNotClicked(button: button.uiButton)
+            
+            let generatedImage = UIImage.createBy(color: UIColor(hex: "606060"), size: button.imageSize!)
```
Comment (liuvasconcelos) [inline StantDrawing/BrushTool/GroupSizeView.swift]: Try not to force stuffs.

File: StantDrawing/BrushTool/GroupSizeView.swift [hunk 6]
```diff
@@ -133,25 +169,36 @@ public class GroupSizeView: UIView {
...
+            
+            button.uiButton.setImage(generatedImage, for: .normal)
+            
+            let difference = (button.uiButton.frame.height - button.imageSize!.height) / 2
```
Comment (liuvasconcelos) [inline StantDrawing/BrushTool/GroupSizeView.swift]: Here as well.

File: StantDrawing/BrushTool/GroupSizeView.swift [hunk 7]
```diff
@@ -133,25 +169,36 @@ public class GroupSizeView: UIView {
...
+                
+            button.uiButton.imageEdgeInsets = UIEdgeInsetsMake(difference,
```
Comment (liuvasconcelos) [inline StantDrawing/BrushTool/GroupSizeView.swift]: For "difference" do you mean something like "margin" or "padding" ? We can make this name more legible.

File: StantDrawing/BrushTool/GroupSizeView.swift [hunk 8]
```diff
@@ -258,15 +305,18 @@ public class GroupSizeView: UIView {
...
+
         selectColorImageView.image = selectColorIcon
-        
-    button.setImage(selectColorImageView.image?.withRenderingMode(.alwaysTemplate), for: .normal)
-        
-        button.tintColor = UIColor(hex: imageColor)
+
+        button.uiButton.setImage(selectColorImageView.image?.withRenderingMode(.alwaysTemplate), for: .normal)
+
+        button.uiButton.heightAnchor.constraint(equalToConstant: button.imageSize!.height).isActive = true
```
Comment (liuvasconcelos) [inline StantDrawing/BrushTool/GroupSizeView.swift]: Try to avoid forcing :)

**Review by liuvasconcelos — CHANGES_REQUESTED** [review]

File: StantDrawing/BrushTool/BrushToolView.swift (hunk 1 above)
Comment (alexandrehcdc) [reply StantDrawing/BrushTool/BrushToolView.swift]: Done :)

File: StantDrawing/BrushTool/BrushToolView.swift (hunk 2 above)
Comment (alexandrehcdc) [reply StantDrawing/BrushTool/BrushToolView.swift]: Which comment? If it is related to the attribution as an if, consider it done :)

File: StantDrawing/BrushTool/GroupSizeView.swift (hunk 3 above)
Comment (alexandrehcdc) [reply StantDrawing/BrushTool/GroupSizeView.swift:58]: Done :)

File: StantDrawing/BrushTool/GroupSizeView.swift (hunk 4 above)
Comment (alexandrehcdc) [reply StantDrawing/BrushTool/GroupSizeView.swift:68]: Done :)

File: StantDrawing/BrushTool/GroupSizeView.swift (hunk 5 above)
Comment (alexandrehcdc) [reply StantDrawing/BrushTool/GroupSizeView.swift]: Done!

File: StantDrawing/BrushTool/GroupSizeView.swift (hunk 6 above)
Comment (alexandrehcdc) [reply StantDrawing/BrushTool/GroupSizeView.swift]: Done!

File: StantDrawing/BrushTool/GroupSizeView.swift (hunk 7 above)
Comment (alexandrehcdc) [reply StantDrawing/BrushTool/GroupSizeView.swift]: Done :)

File: StantDrawing/BrushTool/GroupSizeView.swift (hunk 8 above)
Comment (alexandrehcdc) [reply StantDrawing/BrushTool/GroupSizeView.swift]: Done :)

**Review by liuvasconcelos — APPROVED** [review]

**Pull request merged**